src/vm/hints/hint_utils.rs

```diff
@@ -275,3 +276,76 @@ pub fn assert_le_felt(
+pub fn is_positive(
+    vm: &mut VirtualMachine,
+    ids: HashMap<String, BigInt>,
+) -> Result<(), VirtualMachineError> {
+    //Check that ids contains the reference id for each variable used by the hint
+    let (value_ref, is_positive_ref) = if let (Some(value_ref), Some(is_positive_ref)) = (
+        ids.get(&String::from("value")),
+        ids.get(&String::from("is_positive")),
+    ) {
+        (value_ref, is_positive_ref)
+    } else {
+        return Err(VirtualMachineError::IncorrectIds(
+            vec![String::from("value"), String::from("is_positive")],
+            ids.into_keys().collect(),
+        ));
+    };
+    //Check that each reference id corresponds to a value in the reference manager
+    let (value_addr, is_positive_addr) = if let (Some(value_addr), Some(is_positive_addr)) = (
+        get_address_from_reference(value_ref, &vm.references, &vm.run_context),
+        get_address_from_reference(is_positive_ref, &vm.references, &vm.run_context),
+    ) {
+        (value_addr, is_positive_addr)
+    } else {
+        return Err(VirtualMachineError::FailedToGetIds);
+    };
+    //Check that the ids are in memory
+    match (vm.memory.get(&value_addr), vm.memory.get(&is_positive_addr)) {
+        (Ok(Some(maybe_rel_value)), Ok(_)) => {
+            //Check that the value at the ids address is an Int
+            let value = if let MaybeRelocatable::Int(ref value) = maybe_rel_value {
+                value
+            } else {
+                return Err(VirtualMachineError::ExpectedInteger(value_addr.clone()));
+            };
+            for (name, builtin) in &vm.builtin_runners {
+                //Check that range_check_builtin is present
+                if name == &String::from("range_check") {
+                    let range_check_builtin = if let Some(range_check_builtin) =
+                        builtin.as_any().downcast_ref::<RangeCheckBuiltinRunner>()
+                    {
+                        range_check_builtin
+                    } else {
+                        return Err(VirtualMachineError::NoRangeCheckBuiltin);
+                    };
+                    //Main logic (assert a is positive)
+                    let mut result = bigint!(0);
+                    let int_value = as_int(value, &vm.prime);
+                    if int_value.abs() > range_check_builtin._bound {
+                        return Err(VirtualMachineError::ValueOutsideValidRange(int_value));
+                    }
+                    if int_value > bigint!(0) {
+                        result = bigint!(1);
+                    }
+                    return match vm
+                        .memory
+                        .insert(&is_positive_addr, &MaybeRelocatable::from(result))
+                    {
+                        Ok(_) => Ok(()),
+                        Err(memory_error) => Err(VirtualMachineError::MemoryError(memory_error)),
+                    };
```

Comment:
Not sure if it'd work here, but I suggest trying:
```suggestion
                        .insert(&is_positive_addr, &MaybeRelocatable::from(result))
                        .map_err(VirtualMachineError::MemoryError);
```

Reply:
It works!